lib/tasks/journeys_report.rake

```diff
@@ -82,39 +32,47 @@ namespace :journeys do
             billable: journey.billable,
             events: journey.events.default_order.map do |event|
               {
-                timestamp: event.client_timestamp.strftime('%d/%m %H:%M:%S'),
-                event: event.event_name,
-                notes: event.notes,
-                from_location: event.from_location&.title,
-                to_location: event.to_location&.title,
+                  timestamp: event.client_timestamp.strftime('%d/%m %H:%M:%S'),
+                  event: event.event_name,
+                  notes: event.notes,
+                  from_location: event.from_location&.title,
+                  to_location: event.to_location&.title,
               }
-            end,
+            end
           }
         end
         moves << {
+          supplier: supplier.key,
           reference: move.reference,
           date: move.date,
           from: move.from_location.title,
           to: move.to_location&.title,
           events: move.move_events.default_order.map do |move_event|
-            {
-              timestamp: move_event.client_timestamp.strftime('%d/%m %H:%M:%S'),
-              event: move_event.event_name,
-              notes: move_event.notes,
-              from_location: move_event.from_location&.title,
-              to_location: move_event.to_location&.title,
-            }
-          end,
+                    {
+                      timestamp: move_event.client_timestamp.strftime('%d/%m %H:%M:%S'),
+                      event: move_event.event_name,
+                      notes: move_event.notes,
+                      from_location: move_event.from_location&.title,
+                      to_location: move_event.to_location&.title,
+                    }
+                  end,
           journeys: journeys,
         }
       end
-      suppliers << {
-        supplier: supplier.key,
-        moves: moves,
-      }
-    end
-    File.open('journeys_report.json', 'w') do |file|
-      file.write(JSON.pretty_generate({ data: suppliers }))
+
+      File.open(filename, 'w') do |file|
+        file.write(JSON.pretty_generate({ moves: moves })) # maybe change to .to_json for production
+      end
+
+      puts "Exporting #{filename} to S3..."
+      bucket_name = ENV['S3_REPORTING_BUCKET_NAME']
+      if bucket_name.present?
+        s3 = Aws::S3::Resource.new
+        obj = s3.bucket(bucket_name).object(filename)
+        obj.upload_file(filename)
```

Comment:
```suggestion
        obj.put(body: StringIO.new(json_data))
```

Reply:
Yes, that's a good suggestion.  
I was doing something similar with `obj.put` initially (although it was from the file, which kinda defeated the purpose).

This is an early stage and the file format will likely suffer many mutations, so it is convenient to have the intermediate file for now in pretty print format.

But we surely move to this approach when it's closer to it's final form.

Reply:
yes, I agree the intermediate file format is clearer for now. Keep in mind this is still a proof of concept, where it is important to be able to debug and inspect at any stage, and not a finished system